SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -372,6 +389,73 @@
   <!-- End of FxCop section-->
   <!-- **************************************************************************** -->
 
+  <!-- **************************************************************************** -->
+  <!-- Roslyn analysis section -->
+  <!-- **************************************************************************** -->
+  <!-- Set the properties required to run Roslyn analysis as part of the build. 
+       Note: Roslyn analysis is run using the CSC task in MSBuild 14.0 onwards.
+       Analysis is controlled by the following properties:
+         ResolvedCodeAnalysisRuleSet - the full path to the ruleset to use
+         ErrorLog - the file path for the error reporting file
+         
+       We want to run after the core "ResolveCodeAnalysisRuleSet" target which
+       assigns a value to "ResolvedCodeAnalysisRuleSet".
+  -->
+  <Target Name="OverrideRoslynCodeAnalysisProperties"
+        Condition=" $(SonarQubeTempPath) != '' "
+        DependsOnTargets="SonarQubeCategoriseProject"
+        AfterTargets="ResolveCodeAnalysisRuleSet"
+        BeforeTargets="CoreCompile">
+
+    <PropertyGroup>
+      <!-- Check if the ruleset exists -->
+      <SonarQubeRoslynRulesetFullName>$(SonarQubeConfigPath)SonarQubeRoslyn-cs.ruleset</SonarQubeRoslynRulesetFullName>
+      <SonarQubeRoslynRulesetExists>$([System.IO.File]::Exists($(SonarQubeRoslynRulesetFullName)))</SonarQubeRoslynRulesetExists>
+
+      <!-- Run FxCop only if the ruleset exists (else it means no Roslyn rules are enabled) -->
+      <SonarQubeRunRoslynCodeAnalysis>$(SonarQubeRoslynRulesetExists)</SonarQubeRunRoslynCodeAnalysis>
+
+      <!-- For the time being only run for C# -->
+      <SonarQubeRunRoslynCodeAnalysis Condition=" $(Language) != 'C#' ">false</SonarQubeRunRoslynCodeAnalysis>
+
+      <!-- Don't run FxCop if the project is excluded -->
+      <SonarQubeRunRoslynCodeAnalysis Condition=" $(SonarQubeExclude) == 'true' ">false</SonarQubeRunRoslynCodeAnalysis>
+
+      <!-- Don't run FxCop if the project is a test one -->
+      <SonarQubeRunRoslynCodeAnalysis Condition=" $(SonarQubeTestProject) == 'true' ">false</SonarQubeRunRoslynCodeAnalysis>
+
+    </PropertyGroup>
+
+    <PropertyGroup Condition=" $(SonarQubeRunRoslynCodeAnalysis) != 'true' ">
+      <ErrorLog></ErrorLog>
+      <ResolvedCodeAnalysisRuleSet></ResolvedCodeAnalysisRuleSet>
+
+    </PropertyGroup>
+
+    <PropertyGroup Condition=" $(SonarQubeRunRoslynCodeAnalysis) == 'true' ">
+      <ErrorLog Condition=" $(ErrorLog) == '' " >$(TargetDir)SonarQube.Roslyn.ErrorLog.json</ErrorLog>
+      <ResolvedCodeAnalysisRuleSet>$(SonarQubeRoslynRulesetFullName)</ResolvedCodeAnalysisRuleSet>
+    </PropertyGroup>
+
+  </Target>
+
+  <Target Name="SetRoslynAnalysisResults"
```

Comment:
@duncanpMS shouldn't this target explicitly depend on `OverrideRoslynCodeAnalysisProperties` ? (today all works because of the indirect dependency between `Build` and `CoreCompile`)